### PR TITLE
Prometheus: Fix field name setting for heatmap-cells type

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**

We don't want to change the field name when the response type is `heatmap-cells`. For this type we have `xMax`, `yMin`, `yMax`, `count`, `yLayout` fields. If that part in the code runs for `heatmap-cells` it changes the field name with the `__name__` label value. That is breaking our heatmap visualization. So we skip that part simply adding a condition for it. 

### How to reproduce the error?
- Create a native histogram producing Prometheus datasource i.e. https://github.com/laupse/prometheus-native-histograms
- Create a dashboard and add a panel with a native histogram metric
- Select Histogram visualization
- Select range query
- Run the query

**Why do we need this feature?**

To be able to visualize the native histograms

**Who is this feature for?**

Prometheus native histogram users

### CODE REVIEW

**Review by NWRichmond — COMMENTED** [review]
@itsmylife this looks good, but I'd like to reproduce the issue if possible. On `main`, I'm able to use [heatmaps with native histograms](https://github.com/user-attachments/assets/16b6e8ad-2198-49cd-bd99-246a84ce54cb), and I can't figure out how to break the visualization. Any tips would be appreciated!

File: pkg/promlib/querydata/response.go [hunk 1]
```diff
@@ -113,14 +113,16 @@ func addMetadataToMultiFrame(q *models.Query, frame *data.Frame) {
 	}
 	frame.Fields[0].Config = &data.FieldConfig{Interval: float64(q.Step.Milliseconds())}
 
-	customName := getName(q, frame.Fields[1])
-	if customName != "" {
-		frame.Fields[1].Config = &data.FieldConfig{DisplayNameFromDS: customName}
-	}
+	if frame.Meta.Type != "heatmap-cells" {
```
Comment (kylebrandt) [inline pkg/promlib/querydata/response.go:116]: This condition can return, and then the following code can be dedented.

Comment (itsmylife) [reply pkg/promlib/querydata/response.go]: Please recheck the code. I updated it.

**Comment by itsmylife** [comment]
> @itsmylife this looks good, but I'd like to reproduce the issue if possible. On `main`, I'm able to use [heatmaps with native histograms](https://github.com/user-attachments/assets/16b6e8ad-2198-49cd-bd99-246a84ce54cb), and I can't figure out how to break the visualization. Any tips would be appreciated!

I updated the description with how you can reproduce it.

**Review by NWRichmond — APPROVED** [review]
On `main`, I can reproduce the issue:


missing yMin on main

On `ismail/prom/native-histogram-field-name-fix`, the issue is fixed:


ymin fixed on feature branch

Nicely done!

**Review by kylebrandt — APPROVED** [review]
Might be a good candidate for an e2e test

**Pull request merged**